src/internal/module/types.go

```diff
@@ -0,0 +1,15 @@
+package module
+
+type Version struct {
+	Version string `json:"version"` // The version number of the provider. Correlates to a tag in the module repository
+}
+
+type MetadataFile struct {
+	Versions []Version `json:"versions"`
+}
+
+type Module struct {
+	Namespace string // The module namespace
+	Name      string // The module name
+	System    string // The module system
```

Comment:
Total nit, feel free to ignore: This is called TargetSystem in the `registry-address` package, so we might want to stay consistent with that.

Reply:
Changed to `TargetSystem` - 253b912aed3789846025bb8df3740697476226ea